libs/extract/src/lib/helperFunction.ts

```diff
@@ -10,3 +12,58 @@ export const validateClassName = (value: IndicatorType) => {
       return ''
   }
 }
+
+export const validateInputValue = (
+  name: string,
+  rules: any,
+  value: string,
+  setError: React.Dispatch<React.SetStateAction<Record<string, any>>>
+) => {
+  if (rules?.custom instanceof Function) {
+    const ruleResult: string = rules?.custom()
+    if (ruleResult) {
+      setErrorInsert(setError, name)
+      return 'error'
+    } else {
+      setErrorRemove(setError, name)
+      return
+    }
+  } else {
+    switch (rules.type) {
+      case 'Required': {
+        if (value === '' || value === undefined || value === null) {
+          setErrorInsert(setError, name)
+          return 'error'
+        } else {
+          setErrorRemove(setError, name)
+          return
+        }
+      }
+      default: {
+        return
+      }
+    }
+  }
+}
```

Comment:
looking at the parameter it seemed that the function has many responsibility, maybe we can decouple the error dispatch to the caller and let the function focuses more on validation?

```typescript
export type Rule = {
  custom?: (...args: Array<any>) => string; // expect return value to be error code or empty string
  type?: string;
  // future condition such as min, max, length etc.
}

export const validateInputValue = (name: string, rule: Rule, value: string) => {
  if (rule?.custom) {
    return rules.custom();
  }

  switch (rule?.type) {
    case 'Required': {
      return (value === '' || value === undefined || value === null) ? 'error_code_empty';
    default: 
      return '';
   }
}
```

Reply:
Yeap, this looks way cleaner and readable, will refactor it